torch/optim/swa_utils.py

```diff
@@ -162,12 +164,15 @@ class AveragedModel(Module):
     .. _Polyak averaging:
         https://paperswithcode.com/method/polyak-averaging
     """
-    def __init__(self, model, device=None, avg_fn=None, multi_avg_fn=None, use_buffers=False):
+    def __init__(self, model, device=None, avg_fn=None, multi_avg_fn=None, use_buffers=False, should_deepcopy=True):
         super().__init__()
         assert avg_fn is None or multi_avg_fn is None, 'Only one of avg_fn and multi_avg_fn should be provided'
-        self.module = deepcopy(model)
-        if device is not None:
-            self.module = self.module.to(device)
+        if should_deepcopy:
+            self.module = deepcopy(model)
+            if device is not None:
```

Comment:
It feels like the model should be moved to the device even if it's not deepcopied, no?

Otherwise the device argument and the should_deepcopy should be documented as mutually exclusive.